### PR TITLE
CSR: optionally set delegable hypervisor exceptions

### PR DESCRIPTION
**Pull request by poemonsense**
**Related issue**: 

**Type of change**: bug report

**Impact**: API modification

**Development Phase**: implementation

**Release Notes**

The rocket-chip `medeleg` CSR is intentionally designed to hold only legal values. Given that `usingHypervisor` is used to distinguish whether the hardware supports hypervisor extensions (as seen from the source code everywhere), we should use it for the delegable exceptions as well.

### CODE REVIEW

**Comment by ZenithalHourlyRate** [comment]
Should be made against dev branch and it will be backported.

Have gone through the code and LGTM.

**Comment by poemonsense** [comment]
Should I PR to the dev branch? Or are you going to do it for me?

**Comment by ZenithalHourlyRate** [comment]
> Should I PR to the dev branch?

You can cherry-pick the commit to the dev branch then change the PR target branch

**Comment by poemonsense** [comment]
I think I don't have access to the dev branch.

**Comment by ZenithalHourlyRate** [comment]
https://github.com/chipsalliance/rocket-chip/tree/dev

This one

**Comment by poemonsense** [comment]
But I don't have push access to the rocket-chip repo, including the dev branch. Currently I can only fork rocket-chip and push to the forked one.

**Comment by ZenithalHourlyRate** [comment]
> But I don't have push access to the rocket-chip repo, including the dev branch. Currently I can only fork rocket-chip and push to the forked one.

What I meant is cherry-picking the commit to the dev branch in your forked repo and renaming your fix-delegable-exceptions branch to that branch, then force-push to update the PR here; then update the target branch in this page.

**Comment by poemonsense** [comment]
Done. PR to the dev.

**Review by ZenithalHourlyRate — APPROVED** [review]

**Comment by ZenithalHourlyRate** [comment]
@mergifyio backport master

**Pull request merged**
